src/Engine/Renderer/RenderTechnique/RenderTechnique.cpp

```diff
@@ -120,7 +118,7 @@ static std::map<std::string, DefaultTechniqueBuilder> EngineTechniqueRegistry;
 /** register a new default builder for a technique
  *  @return true if builder added, false else (e.g, a builder with the same name exists)
  */
-bool registerDefaultTechnique( const std::string& name, DefaultTechniqueBuilder builder ) {
+bool registerDefaultTechnique(const std::string& name, const DefaultTechniqueBuilder &builder ) {
```

Comment:
Why here don't you pass `name` by copy as well ? This is not consistent with the other changes.

Reply:
Passing by value and moving is interesting for constructor only, when initializing member data. Then, the object is more efficiently constructed (well, thats depend on the compiler) and initialized. Here, it is not a constructor, so we need a reference to avoid a copy.
Reminds that when initializing an object, a copy is mandatory for initializing member data - so value + std::move = 1 copy directly in the member data. For others, such a copy is not mandatory - so const ref = no copy exept if explicitly needed in the method.
This is how I understand the object construction mechanism and the move semantic.

Reply:
My bad, I though I had seen  a change involving a move for a method as well, but I cannot find it again, I might have wrongly read your code.